Simulator/Core/Simulator.cpp

```diff
@@ -197,10 +198,11 @@ void Simulator::advanceEpoch(const int &currentEpoch) const
    uint64_t count = 0;
    // Compute step number at end of this simulation epoch
    uint64_t endStep = g_simulationStep + static_cast<uint64_t>(epochDuration_ / deltaT_);
+   model_->getLayout().getVertices().loadEpochInputs(g_simulationStep, endStep);
    // DEBUG_MID(model->logSimStep();) // Generic model debug call
    while (g_simulationStep < endStep) {
       // Output status once every 10,000 steps
```

Comment:
This is something I have been meaning to ask you about. I assume that `deltaT_` is supposed to be a simulation-specific parameter, meaning that it should be part of the config file, but we have always used the `DEFAULT_dt` value from `Global.h` (1e-4).

Since I didn't want to mess with the code in the main loop, I kept the default deltaT_ but assumed every clock tick is one second (1e-4 = 1 second). This still prints to the console every 10,000 clock ticks but every tick is 1 second (about 2.77 hours). I calculated the values to simulate about one month because I am using the one month of call data that we have, for testing purposes.

I realize that this is not the proper way of doing it. Now, should the printing interval also be a simulation parameter, perhaps with a default value when one is not present in the config file?

Reply:
I think the actual deltaT should be set to what it really is and used that way. Right now, we just use the default, but you could add it as a simulation config parameter for the ESCS simulations, with an issue to examine that for the other simulations.

Regarding printing interval, I don't know. It seems excessive to make it a simulation parameter. In an ideal world, it would be based on actual clock time, so updates come every _n_ seconds; whatever is best from a user interface point of view. Setting that aside, maybe we should calculate the total number of time steps in a simulation and set the output interval to be something like 1% of simulation length.

Reply:
I will address both of these in this PR. Printing every 1% percent of the simulation length seems reasonable, for now.

BTW, what is the actual clock time of one simulation step in the Neural Network model?

Reply:
It depends on the hardware it's running on, right? To do it that way, we'd need to get the actual time periodically and output when the elapsed time is some amount beyond whatever the update interval is, in seconds or minutes. Not super complicated, but not worth you spending time one. A good piece of an undergrad student capstone.

Reply:
My bad, I meant to ask what the simulation time is per simulation step. What is the unit used for `epochDuration_`?